doc/userguide/devguide/codebase/installation-from-git.rst

```diff
@@ -0,0 +1,165 @@
+  libcap-ng0 make libmagic-dev libjansson-dev rustc cargo jq git-core
+
+Add ``${HOME}/.cargo/bin`` to your path:
+
+.. code-block:: bash
+
+  export PATH=$PATH:${HOME}/.cargo/bin
+  cargo install --force cbindgen
+
+Depending on the current status of your system, it may take a while to
+complete this process.
+
+**IPS**
+
+By default, Suricata works as an IDS. If you want to use it as an IDS and IPS
+program, enter:
+
+.. code-block:: bash
+
+  sudo apt-get -y install libnetfilter-queue-dev libnetfilter-queue1 \
+  libnfnetlink-dev libnfnetlink0
+
+Suricata
+~~~~~~~~
+
+First, it is convenient to create a directory for Suricata.
+Name it 'suricata' or 'oisf', for example. Open the terminal and enter:
+
+.. code-block:: bash
+
+  mkdir suricata  # mkdir oisf
+
+Followed by:
+
+.. code-block:: bash
+
+  cd suricata  # cd oisf
+
+Next, enter the following line in the terminal:
+
+.. code-block:: bash
+
+  git clone git@github.com:OISF/suricata.git
+  cd suricata
+
+Libhtp is not bundled. Get libhtp by doing:
+
+.. code-block:: bash
+
+  ./scripts/bundle.sh libhtp
+
+Followed by:
+
+.. code-block:: bash
+
+  ./autogen.sh
+
+
+To configure, please enter:
+
+.. code-block:: bash
+
+  ./configure
+
+
+To compile, please enter:
+
+.. code-block:: bash
+
+  make
+
+To install Suricata, enter:
+
+.. code-block:: bash
+
+  sudo make install
+  sudo ldconfig
+
+To install suricata-update
+
+Follow the instructions found in
+https://suricata-update.readthedocs.io/en/latest/quickstart.html
+
+.. note:: If you would like to build ``suricata-update`` from source, enter:
+
+  .. code-block:: bash
+
+    ./scripts/bundle.sh suricata-update
+    cd suricata-update
+    python setup.py build
+    sudo python setup.py install
+    sudo suricata-update
+
+Auto-setup
+~~~~~~~~~~
+
+You can also use the available auto-setup features of Suricata. Ex:
+
+.. code-block:: bash
+
+  ./configure && make && sudo make install-conf
+
+*make install-conf*
+would do the regular "make install" and then it would automatically
+create/setup all the necessary directories and ``suricata.yaml`` for you.
+
+.. code-block:: bash
+
+  ./configure && make && make install-rules
+
+*make install-rules*
+would do the regular "make install" and then it would automatically download
+and set-up the latest ruleset from Emerging Threats available for Suricata.
+
+.. code-block:: bash
+
+  ./configure && make && make install-full
+
+*make install-full*
+would combine everything mentioned above (install-conf and install-rules) -
+and will present you with a ready to run (configured and set-up) Suricata.
+
+Post installation
+~~~~~~~~~~~~~~~~~
+
+Please continue with :ref:`Basic setup`.
+
+In case you have already created your Suricata directory and cloned the
+repository, if you want to update your local repository with the most
+recent code, enter the Suricata directory and run:
+
+.. code-block:: bash
+
+  cd suricata/suricata
```

Comment:
This should just be `cd suricata` (if your steps are being followed as that is where `git` cloned to).

Reply:
I reworded it again...